Neos.Fusion/Classes/Package.php

```diff
@@ -66,27 +66,12 @@ public function boot(Bootstrap $bootstrap)
                     $cacheManager = $bootstrap->getEarlyInstance(CacheManager::class);
                     $listener = new FileMonitorListener($cacheManager);
                     $dispatcher->connect(FileMonitor::class, 'filesHaveChanged', $listener, 'flushContentCacheOnFileChanges');
-                    // Use a closure to invoke the FusionParserCache, so the object is not instantiated during compiletime and has working DI
-                    $flushParsePartialsCache = function ($identifier, $changedFilesAndStatus) use ($bootstrap) {
-                        if ($identifier !== 'Fusion_Files') {
-                            return;
-                        }
-                        $objectManager = $bootstrap->getObjectManager();
-                        if ($objectManager->isRegistered(ParserCache::class) === false) {
-                            // if we make a total `rm -rf Data/Temporary` all monitored `*.fusion` files will be seen as newly created.
-                            // this triggers pretty early this `filesHaveChanged` and we still have the CompileTimeObjectManager
-                            // we would get an exception like:
-                            // Cannot build object "FusionParserCache" because it is unknown to the compile time Object Manager.
-                            // so instead we will just make sure the cache is totally flushed.
-                            $cacheManager = $bootstrap->getEarlyInstance(CacheManager::class);
-                            $fusionParsePartialsCache = $cacheManager->getCache('Neos_Fusion_ParsePartials');
-                            $fusionParsePartialsCache->flush();
-                            return;
-                        }
-                        $fusionParserCache = $objectManager->get(ParserCache::class);
-                        $fusionParserCache->flushFileAstCacheOnFileChanges($changedFilesAndStatus);
-                    };
-                    $dispatcher->connect(FileMonitor::class, 'filesHaveChanged', $flushParsePartialsCache);
+
+                    $parsePartialsCache = $cacheManager->getCache('Neos_Fusion_ParsePartials');
+                    // We cannot use the `ParserCache`, not even with a static method, because once a class is registered at boot time,
+                    // It can't be unregistered, so we can never use the Proxy version with DI.
+                    $compileTimeParserCache = new CompileTimeParserCache($parsePartialsCache);
+                    $dispatcher->connect(FileMonitor::class, 'filesHaveChanged', $compileTimeParserCache, 'flushParsePartialsOnFileChanges');
```

Comment:
How about injecting not the actual parser cache but wrapping the flushParsePartialsOnFileChanges call in a closure?

```
        $dispatcher->connect(FileMonitor::class, 'filesHaveChanged', function () use ($bootstrap) {
            $parsePartialsCache = $bootstrap->getObjectManager()->get(Cache\CacheManager::class)->getCache('Neos_Fusion_ParsePartials');
            $parsePartialsCache->flushParsePartialsOnFileChanges();
        });
```

Reply:
thanks for your input i appreciate it!

so basically an extended ParsePartialsVariableFrontent?


or make a popo to just generate the cache ids and use it here in the closure as well as in the `ParserCache` wrapper ...